src/odemis/driver/smaract.py

```diff
@@ -3084,3 +3085,1139 @@ def SA_CTL_Move(self, handle, ch, pos_pm, _):
+
+        # Connection
+        if locator == "fake":
+            self.core = FakePicoscale_DLL()
+        else:
+            self.core = SA_SIDLL()
+        self._locator = locator
+        self._id = self._openConnection(locator)
+
+        # Device information
+        devname = self.GetProperty_s(SA_SIDLL.SA_SI_DEVICE_NAME_PROP)
+        sn = self.GetProperty_s(SA_SIDLL.SA_SI_DEVICE_SERIAL_NUMBER_PROP)
+        num_ch = self.GetNumberOfChannels()
+        self._hwVersion = "SmarAct %s (s/n %s) with %s channels." % (devname, sn, num_ch,)
+        self._swVersion = self.GetFullVersionString()
+        logging.debug("Using Picoscale library version %s to connect to %s. ", self._swVersion, self._hwVersion)
+
+        # Device setup
+        self.EnableFullAccess()
+        try:
+            self.LoadConfiguration()
+        except SA_SIError as ex:
+            raise ValueError("Failed to load configuration. This might indicate that the device is not "
+                             "configured properly. Error: %s." % ex)
+
+        # Log configuration
+        flen = self.GetFiberLength()
+        ext_flen = self.GetExtensionFiberLength()
+        wdist_min, wdist_max = self.GetWorkingDistance()
+        logging.debug("Picoscale configuration: fiber length %s, extension fiber length %s, "
+                      "working distance [%s, %s].", flen, ext_flen, wdist_min, wdist_max)
+
+        # Position polling
+        self.position = model.VigilantAttribute({}, getter=self._getPosition, readonly=True)
+        self._polling_thread = util.RepeatingTimer(1, self._updatePosition, "Position polling")
+        self._polling_thread.start()
+
+        # Precision mode
+        # The precision mode is a special feature that needs to be purchased separately, so it is not available
+        # by default.
+        # TODO: this functionality has not yet been tested
+        if precision_mode != 0:
+            if not self.IsFeatureEnabled(SA_SIDLL.FEATURE_PRECISION_MODE):
+                raise ValueError("Device does not have precision mode. This feature needs to be purchased separately.")
+            self.SetPrecisionMode(precision_mode)
+
+        # State: starting until first referencing/validation procedure is done
+        # Position requests can be made during startup, however, the values are not going to be accurate until the
+        # initialization process is complete. The position thread is polling from the start.
+        # If during startup a call to .reference is made, the referencing future will be added to the
+        # executor and be carried out right after the initialization referencing is done.
+        # In conclusion, no special handling is required during startup, although it does not make much sense
+        # to use the driver until the startup is complete. The position getter will log a warning in this case.
+        self.state._set_value(model.ST_STARTING, force_write=True)
```

Comment:
I think it's nice to have some indication in the software of whether or not the device has finished its initial referencing procedure. It's particularly useful for the testcases, where I just wait until the state is "running". Waiting for "referenced" doesn't work because we have this distinction between validation and referencing and in case of validation I decided not to consider the channel referenced after the procedure is finished.

Reply:
Indeed. Let's use this code. We might want to make this behaviour more officially supported in the back-end, and then use it in other drivers which do referencing at init.